Sources/PackageModel/UserToolchain.swift

```diff
@@ -328,7 +328,11 @@ public final class UserToolchain: Toolchain {
         swiftSDK: SwiftSDK,
         environment: EnvironmentVariables
     ) throws -> [String] {
-        let swiftCompilerFlags = swiftSDK.toolset.knownTools[.swiftCompiler]?.extraCLIOptions ?? []
+        var swiftCompilerFlags = swiftSDK.toolset.knownTools[.swiftCompiler]?.extraCLIOptions ?? []
+
+        if let linker = swiftSDK.toolset.knownTools[.linker]?.path {
+            swiftCompilerFlags += ["-use-ld=\(linker._nativePathString(escaped: false))"]
```

Comment:
As https://github.com/apple/swift-package-manager/pull/6730 is on track to be merged, I don't think that either explicit `pathString` or `_nativePathString` are necessary here. As I'm extending `DefaultStringInterpolation` for `AbsolutePath` in that PR, just passing the path directly should be enough:

```suggestion
            swiftCompilerFlags += ["-use-ld=\(linker)"]
```

Reply:
Sounds good, didn't want the build to be broken prior to #6730 being merged but I now realize that `AbsolutePath` already conforms to `CustomStringConvertible` so interpolation should already mostly work

Reply:
Oh, I forgot to add quotes to account for files with spaces, updated the suggestion. @compnerd do you think we should actually add quotes in that `DefaultStringInterpolation`? I assume we'll benefit from quotes in all contexts where paths can be interpolated into a string, won't we?

Reply:
I tested out paths with spaces and they work even without quotes — the argument is ultimately passed directly to `posix_spawn`/`execv`/etc (vis-a-vis a shell) so there's no need for escaping afaict

Reply:
Ok, even though we don't use `posix_spawn`/`execv` on Windows, seems like quoting is implemented on Foundation side anyway, then not quoting it here looks ok: https://github.com/apple/swift-corelibs-foundation/blob/adf511f1bdd5c1a205c5238c166ecd667ef8a60a/Sources/Foundation/Process.swift#L163